embeddings.go

```diff
@@ -241,11 +252,29 @@ func (c *Client) CreateEmbeddings(
 	conv EmbeddingRequestConverter,
 ) (res EmbeddingResponse, err error) {
 	baseReq := conv.Convert()
+
+	// Prepare the body with only the provided fields.
+	// The body map is used to dynamically construct the request payload for the embedding API.
+	// Instead of relying on a fixed struct, the body map allows for flexible inclusion of fields
+	// based on their presence, avoiding unnecessary or empty fields in the request.
+	body := make(map[string]any)
+	body["input"] = baseReq.Input
+	body["model"] = baseReq.Model
+	if baseReq.User != "" {
+		body["user"] = baseReq.User
+	}
+	if baseReq.EncodingFormat != "" {
+		body["encoding_format"] = baseReq.EncodingFormat
+	}
+	if baseReq.Dimensions > 0 { // Assuming 0 means the field is not set
+		body["dimensions"] = baseReq.Dimensions
+	}
```

Comment:
Thank you for the PR! Do I understand correctly that we'll need to extend logic here every time `EmbeddingRequest` gets updated?

Reply:
If so, could we maybe serialize `baseReq` to json and than back to `map[string]any`? That would allow us not to keep this logic updated

Reply:
yes will need to extend logic here also if EmbeddingRequest gets updated. But yeah I think to support extra_body param this approach is the way to do so in Golang.

Reply:
> If so, could we maybe serialize `baseReq` to json and then back to `map[string]any`? That would allow us not to keep this logic updated

Okay will look into it and update the PR

Reply:
@sashabaranov I have updated the PR and tested the changes, can you please look into the changes?